x/staking/types/delegation.go

```diff
@@ -112,24 +114,50 @@ func (e UnbondingDelegationEntry) IsMature(currentTime time.Time) bool {
 	return !e.CompletionTime.After(currentTime)
 }
 
+// OnHold - is the current entry on hold due to external modules
```

Comment:
can you touch on why this is needed?

Reply:
Normally, unbonding operations (e.g., unbonding delegations) need to wait for `UnbondingPeriod` to elapse before they can be completed (e.g., the delegated stake to be returned to the delegator). This is to allow for evidence of misbehavior to be submitted and for the validators to get slashed. 

In the case of Interchain Security, validators can misbehave also on the consumer chain. That means that unbonding operations should not complete before the `UnbondingPeriod` elapses also on the consumer chains. As a result, an unboding operation can complete if
- it is matured, i.e., `UnbondingPeriod` has elapsed on the provider chain;
- it is not on hold, i.e., an external module (e.g., the CCV module) no longer keeps the operation on hold (e.g., the CCV module received notifications that the `UnbondingPeriod` has elapsed on all consumer chains).